scripts/interactive_tests/interactive_test1.js

```diff
@@ -0,0 +1,33 @@
+var InteractiveTest = require('./interactive_test_util').InteractiveTest;
+var port = 6969;
+var test = new InteractiveTest('node lib/cli.js --elementExplorer true', port);
+
+// Check state persists.
+test.addCommandExpectation('var x = 3');
+test.addCommandExpectation('x', '3'); 
+
+// Check can return functions.
+test.addCommandExpectation('var y = function(param) {return param;}');
+test.addCommandExpectation('y', 'function (param) {return param;}'); 
+
+// Check promises complete.
+test.addCommandExpectation('browser.driver.getCurrentUrl()', 'data:,');
+test.addCommandExpectation('browser.get("http://localhost:8081")');
+test.addCommandExpectation('browser.getCurrentUrl()', 
+    'http://localhost:8081/#/form'); 
+
+// Check promises are resolved before being returned.
+test.addCommandExpectation('var greetings = element(by.binding("greeting"))');
+test.addCommandExpectation('greetings.getText()', 'Hiya');
+
```

Comment:
Can we add a test that `element(by.binding("nonexistent"))` gives a reasonable error message instead of blowing up?